docs: refine code splitting guide (#4969)
rolldown-rs/rolldown: docs/guide/in-depth/code-splitting.md

### Respecting Execution Order doesn't take precedence

> However, Rolldown sometimes places modules without respecting their original order. This is because ensuring modules to be singleton takes precedence over placing modules in the order they are declared.

Author: ```suggestion
However, Rolldown sometimes places modules without respecting their original order. This is because ensuring that modules are singletons takes precedence over placing modules in the order they are declared.
```

> However, Rolldown sometimes places modules without respecting their original order. This is because ensuring that modules are singletons takes precedence over placing modules in the order they are declared.

Author: ```suggestion
However, Rolldown sometimes places modules without respecting their original order. This is because ensuring that modules are singletons takes precedence over placing them in the declared order.
```
